### PR TITLE
UCP/API/TEST: Run UCP tests on multiple transports.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/833/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1446/ for details.

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -373,6 +373,20 @@ void ucp_config_release(ucp_config_t *config);
 
 /**
  * @ingroup UCP_CONFIG
+ * @brief Modify context configuration.
+ *
```
Comment (shamisp) [inline src/ucp/api/ucp.h]: please include full documentation besides @brief

File: test/gtest/ucp/ucp_test.h [hunk 2]
```diff
@@ -46,15 +52,64 @@ class ucp_test : public ucs::test_base, public ::testing::Test {
...
+ * Instantiate the parameterized test case for all transport combinations.
+ *
+ * @param _test_case  Test case class, derived from uct_test.
+ */
+#define UCP_INSTANTIATE_TEST_CASE(_test_case) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, ud,    "ud"                ) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, udrc,  "ud", "rc"          ) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, shm,   "mm", "knem"        ) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, udrcx, "ud_mlx5", "rc_mlx5") \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, ugni,  "ugni_smsg", "ugni_udt", "ugni_rdma")
```
Comment (shamisp) [inline test/gtest/ucp/ucp_test.h]: I don't see CMA, XPMEM

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/878/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1483/ for details.

**Comment by shamisp** [comment]
:+1:

**Pull request merged**